contrib/kind.sh

```diff
@@ -21,29 +21,103 @@ run_kubectl() {
   done
 }
 
+# Set default values
+KIND_CLUSTER_NAME=${KIND_CLUSTER_NAME:-ovn}
 K8S_VERSION=${K8S_VERSION:-v1.17.2}
 KIND_INSTALL_INGRESS=${KIND_INSTALL_INGRESS:-false}
 KIND_HA=${KIND_HA:-false}
 if [ "$KIND_HA" == true ]; then
   DEFAULT_KIND_CONFIG=./kind-ha.yaml
 else
-  DEFAULT_KIND_CONFIG=./kind.yaml
+  DEFAULT_KIND_CONFIG=./kind-noha.yaml
 fi
 KIND_CONFIG=${KIND_CONFIG:-$DEFAULT_KIND_CONFIG}
 KIND_REMOVE_TAINT=${KIND_REMOVE_TAINT:-true}
+IPV4_SUPPORT=${IPV4_SUPPORT:-true}
+IPV6_SUPPORT=${IPV6_SUPPORT:-false}
+
+# Input not currently validated. Modify outside script at your own risk.
+NET_CIDR_IPV4=${NET_CIDR_IPV4:-10.244.0.0/16}
+SVC_CIDR_IPV4=${SVC_CIDR_IPV4:-10.96.0.0/12}
+NET_CIDR_IPV6=${NET_CIDR_IPV6:-fd01::/24}
+SVC_CIDR_IPV6=${SVC_CIDR_IPV6:-fd02::/24}
+
+# Make a copy of the kind-xxx.yaml file so it can be edited in place.
+KIND_CONFIG_LCL=kind.yaml
+cp ${KIND_CONFIG} ${KIND_CONFIG_LCL}
 
 # Detect IP to use as API server
-API_IP=$(ip -4 addr | grep -oP '(?<=inet\s)\d+(\.\d+){3}' | grep -v "127.0.0.1" | head -n 1)
-if [ -z "$API_IP" ]; then
-  echo "Error detecting machine IP to use as API server"
+API_IPV4=""
+if [ "$IPV4_SUPPORT" == true ]; then
+  # ip -4 addr -> Run ip command for IPv4
+  # grep -oP '(?<=inet\s)\d+(\.\d+){3}' -> Use only the lines with the
+  #   IPv4 Addresses and strip off the trailing subnet mask, /xx
+  # grep -v "127.0.0.1" -> Remove local host
+  # head -n 1 -> Of the remaining, use first entry
+  API_IPV4=$(ip -4 addr | grep -oP '(?<=inet\s)\d+(\.\d+){3}' | grep -v "127.0.0.1" | head -n 1)
+  if [ -z "$API_IPV4" ]; then
+    echo "Error detecting machine IPv4 to use as API server"
+    exit 1
+  fi
+fi
+
+API_IPV6=""
+if [ "$IPV6_SUPPORT" == true ]; then
+  # ip -6 addr -> Run ip command for IPv6
+  # grep "inet6" -> Use only the lines with the IPv6 Address
+  # sed 's@/.*@@g' -> Strip off the trailing subnet mask, /xx
+  # grep -v "::1" -> Remove local host
+  # sed '/^fe80:/ d' -> Remove Link-Local Addresses
+  # head -n 1 -> Of the remaining, use first entry
+  API_IPV6=$(ip -6 addr  | grep "inet6" | awk -F' ' '{print $2}' | \
+             sed 's@/.*@@g' | grep -v "::1" | sed '/^fe80:/ d' | head -n 1)
+  if [ -z "$API_IPV6" ]; then
+    echo "Error detecting machine IPv6 to use as API server"
+    exit 1
+  fi
+fi
+
+if [ "$IPV4_SUPPORT" == true ] && [ "$IPV6_SUPPORT" == false ]; then
+  API_IP=${API_IPV4}
+  NET_CIDR=$NET_CIDR_IPV4
+  SVC_CIDR=$SVC_CIDR_IPV4
+  echo "IPv4 Only Support: API_IP=$API_IP --net-cidr=$NET_CIDR --svc-cidr=$SVC_CIDR"
+elif [ "$IPV4_SUPPORT" == false ] && [ "$IPV6_SUPPORT" == true ]; then
+  API_IP=${API_IPV6}
+  NET_CIDR=$NET_CIDR_IPV6
+  SVC_CIDR=$SVC_CIDR_IPV6
+  echo "IPv6 Only Support: API_IP=$API_IP --net-cidr=$NET_CIDR --svc-cidr=$SVC_CIDR"
+
+  # Bail for now
+  echo "Invalid setup. IPv6 single stack not currently supported."
```

Comment:
why is single stack ipv6 not supported?

Reply:
Is it supported? I can fix this. I just thought it wasn't supported yet.

Reply:
In latest merge I allowed it. It failed for me, but didn't spend much time on it. Wasn't sure if it was because my host network IPv4 or something else. I'll follow-up on IPv6 only soon.

Reply:
I updated `contrib/kind.sh`, `contrib/kind.yaml.j2` and `dist/images/ovnkube.sh` and now IPv6 only runs if firewalld is stopped on host.

Reply:
Done

Reply:
thanks, next step is CI job for this